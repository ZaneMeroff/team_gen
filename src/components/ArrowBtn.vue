<template>

  <button
    @click="emitArrowBtnClick"
    :class="arrowDirection === 'up' || arrowDirection === 'down' ? 'arrow-btn' : 'arrow-btn nav-arrow-btn'">
    <span v-html="arrowDisplay"></span>
  </button>

</template>

<script>
  export default {
    name: "ArrowBtn",
    props: {
      arrowDirection: { type: String, default: "up" },
    },
    components: {},
    data() {
      return {
        arrowDisplay: "",
        arrowKey: {
          up:    "&#8593",
          down:  "&#8595",
          left:  "&#x2190",
          right: "&#x2192",
        },
      }
    },
    methods: {

      setArrowDisplay(direction) {
        this.arrowDisplay = this.arrowKey[direction]
      },

      emitArrowBtnClick() {
        this.$emit("arrowClick")
      },
    },
    mounted() {
      this.setArrowDisplay(this.arrowDirection)
    },
    computed: {},
    watch: {},
  }
</script>

<style scoped>

  .arrow-btn {
    height: 75px;
    width: 75px;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 3rem;
  }

  .nav-arrow-btn {
    border-radius: 37.5px;
    padding-bottom: 7px;
  }

  .arrow-btn:hover {
    cursor: pointer;
  }
</style>
