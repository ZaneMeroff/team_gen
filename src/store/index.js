import Vue from "vue"
import Vuex from "vuex"

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    teamNum: 2,
    playerNum: 2,
    playerList: ["", ""],
  },

  mutations: {

    plusteamNum: (state) => state.teamNum++,
    minusteamNum: (state) => state.teamNum !== 2 && state.teamNum--,

    plusplayerNum: (state) => state.playerNum++,
    minusplayerNum: (state) => state.playerNum !== 2 && state.playerNum--,

    updatePlayerList: (state, data) => state.playerList = data,
    updatePlayerName: (state, data) => state.playerList[data.id] = data.name,
  },

  actions: {},
  modules: {},
})
