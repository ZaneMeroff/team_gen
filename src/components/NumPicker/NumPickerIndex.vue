<template>
  <section class="npi-perimeter-container">

    <NumPickerNumDisplay
      :number="pickerType === 'team' ? teamNum : playerNum"
    />

    <div class="arrow-btn-container">
      <ArrowBtn
        @arrowClick="handleArrowClick('up')"
        :arrowDirection="'up'"
      />
      <ArrowBtn
        @arrowClick="handleArrowClick('down')"
        :arrowDirection="'down'"
      />
    </div>

  </section>
</template>

<script>
  import NumPickerNumDisplay from "./NumPickerNumDisplay"
  import ArrowBtn from "../ArrowBtn"

  export default {
    name: "NumPickerIndex",
    props: {
      pickerType: { type: String, default: "team" },
    },
    components: {
      NumPickerNumDisplay,
      ArrowBtn,
    },
    methods: {

      handleArrowClick(payload) {
        if (payload === "up") {
          this.$store.commit(`plus${this.pickerType}Num`)
        } else if (payload === "down") {
          this.$store.commit(`minus${this.pickerType}Num`)
        }
      },
    },
    computed: {

      teamNum() {
        return this.$store.state.teamNum
      },

      playerNum() {
        return this.$store.state.playerNum
      },
    },
  }
</script>

<style scoped>

  .npi-perimeter-container {
    height: 150px;
    width: 150px;
    border: 1px solid #000;
    display: flex;
  }

  .arrow-btn-container {
    display: block;
  }
</style>
