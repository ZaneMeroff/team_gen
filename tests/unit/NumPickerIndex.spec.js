import { createLocalVue, mount, shallowMount } from '@vue/test-utils'
import store from "@/store/index"
import Vuex from "vuex"

import NumPickerIndex from "@/components/NumPicker/NumPickerIndex"
import ArrowBtn from "@/components/ArrowBtn"

// -------- Vuex Setup ------------

const localVue = createLocalVue()
localVue.use(Vuex)

// --------------------------------

describe("NumPickerIndex", () => {

  describe("snapshots", () => {

    it("should render with default props", () => {
      const component = shallowMount(NumPickerIndex, { store, localVue })

      expect(component.element).toMatchSnapshot()
    })
  })

  describe("props", () => {

    describe("pickerType", () => {

      it("should have a default value", () => {
        const component = shallowMount(NumPickerIndex, { store, localVue })

        expect(component.vm.pickerType).toEqual("team")
      })

      it("should accept a string", () => {
        const propsData = { pickerType: "player" }
        const component = shallowMount(NumPickerIndex, {
          propsData, store, localVue
        })

        expect(component.vm.pickerType).toEqual(propsData.pickerType)
      })
    })
  })

  describe("methods", () => {

    describe("handleArrowClick", () => {

      it("should increase teamNum in store if payload is 'up'", () => {
        const propsData = { pickerType: "team" }
        const component = shallowMount(NumPickerIndex, {
          propsData, store, localVue
        })
        const before = store.state.teamNum

        component.vm.handleArrowClick("up")

        const after = store.state.teamNum

        expect(before).toEqual(2)
        expect(after).toEqual(3)
      })

      it("should increase playerNum in store if payload is 'up'", () => {
        const propsData = { pickerType: "player" }
        const component = shallowMount(NumPickerIndex, {
          propsData, store, localVue
        })
        const before = store.state.playerNum

        component.vm.handleArrowClick("up")

        const after = store.state.playerNum

        expect(before).toEqual(2)
        expect(after).toEqual(3)
      })

      it("should decrease teamNum in store if payload is 'down'", () => {
        const propsData = { pickerType: "team" }
        const component = shallowMount(NumPickerIndex, {
          propsData, store, localVue
        })
        store.state.teamNum = 5
        const before = store.state.teamNum

        component.vm.handleArrowClick("down")

        const after = store.state.teamNum

        expect(before).toEqual(5)
        expect(after).toEqual(4)
      })

      it("should decrease playerNum in store if payload is 'down'", () => {
        const propsData = { pickerType: "player" }
        const component = shallowMount(NumPickerIndex, {
          propsData, store, localVue
        })
        store.state.playerNum = 5
        const before = store.state.playerNum

        component.vm.handleArrowClick("down")

        const after = store.state.playerNum

        expect(before).toEqual(5)
        expect(after).toEqual(4)
      })

      it("should NOT decrease teamNum in store if value is 2", () => {
        const propsData = { pickerType: "team" }
        const component = shallowMount(NumPickerIndex, {
          propsData, store, localVue
        })
        store.state.teamNum = 2
        const before = store.state.teamNum

        component.vm.handleArrowClick("down")

        const after = store.state.teamNum

        expect(before).toEqual(2)
        expect(after).toEqual(2)
      })

      it("should NOT decrease playerNum in store if value is 2", () => {
        const propsData = { pickerType: "player" }
        const component = shallowMount(NumPickerIndex, {
          propsData, store, localVue
        })
        store.state.playerNum = 2
        const before = store.state.playerNum

        component.vm.handleArrowClick("down")

        const after = store.state.playerNum

        expect(before).toEqual(2)
        expect(after).toEqual(2)
      })
    })
  })

  describe("computed properties", () => {

    describe("teamNum", () => {

      it("should return default teamNum of 2", () => {
        const component = mount(NumPickerIndex, { store, localVue })

        const expected = 2
        expect(component.vm.teamNum).toEqual(expected)
      })
    })

    describe("playerNum", () => {

      it("should return default playerNum of 2", () => {
        const component = mount(NumPickerIndex, { store, localVue })

        const expected = 2
        expect(component.vm.playerNum).toEqual(expected)
      })
    })
  })

  describe("user interactions", () => {

    describe("events", () => {

      it("should call handleArrowClick when arrowClick is emitted", () => {
        const component = mount(NumPickerIndex, { store, localVue })
        const spy = jest.spyOn(component.vm, "handleArrowClick")
        const arrowBtn = component.findComponent(ArrowBtn)

        arrowBtn.vm.$emit("arrowClick")

        expect(spy).toHaveBeenCalledTimes(1)
      })
    })
  })
})
