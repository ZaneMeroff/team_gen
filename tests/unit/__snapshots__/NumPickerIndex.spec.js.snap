// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumPickerIndex snapshots should render with default props 1`] = `
<section
  class="npi-perimeter-container"
>
  <numpickernumdisplay-stub
    number="2"
  />
   
  <div
    class="arrow-btn-container"
  >
    <arrowbtn-stub
      arrowdirection="up"
    />
     
    <arrowbtn-stub
      arrowdirection="down"
    />
  </div>
</section>
`;
